packages/components/src/systems/RecentOrders/RecentOrdersDuplicateDialog.tsx

```diff
@@ -8,24 +8,27 @@ import { dispatchDatadogAction } from '../../utils/dispatchDatadogAction';
 import formatRxString from '../../utils/formatRxString';
 
 export default function RecentOrdersDuplicateDialog() {
-  let ref: Ref<any> | undefined;
   const [state, actions] = useRecentOrders();
+  const [exit, setExit] = createSignal(false);
 
   createEffect(() => {
     if (state.isDuplicateDialogOpen) {
-      dispatchDatadogAction('prescribe-duplicate-dialog-open', {}, ref);
+      dispatchDatadogAction('prescribe-duplicate-dialog-open', {});
+    }
+  });
+
+  createEffect(() => {
+    if (exit()) {
+      dispatchDatadogAction('prescribe-duplicate-dialog-exit', {});
     }
   });
 
   return (
     <Dialog
       open={state.isDuplicateDialogOpen}
-      onClose={() => {
-        dispatchDatadogAction('prescribe-duplicate-dialog-exit', {}, ref);
-        actions.setIsDuplicateDialogOpen(false);
-      }}
+      onClose={() => actions.setIsDuplicateDialogOpen(false)}
```

Comment:
...but if I try to trigger `prescribe-duplicate-dialog-exit` here it will fail and say it doesn't have access to the context.